fix: handle nullable LCH coordinates after colorjs.io v0.6 upgrade

## Problem

The Renovate PR #197 (`renovate/all-minor-patch`) updates `colorjs.io` from v0.5.2 to v0.6.1. This causes the Preview CI build to fail with TypeScript errors:

```
src/utils/color.ts(8,7): error TS18047: 'c.lch.l' is possibly 'null'.
src/utils/color.ts(12,7): error TS18047: 'c.lch.l' is possibly 'null'.
src/utils/color.ts(13,7): error TS18047: 'c.lch.c' is possibly 'null'.
```

**Failed run:** https://github.com/gekichumai/dxrating/actions/runs/24282493637

## Root Cause

In `colorjs.io` v0.6, LCH color space coordinates (`l`, `c`, `h`) are now typed as `number | null` instead of `number`, aligning with the CSS Color spec where achromatic colors have undefined chroma/hue. The existing code in `apps/web/src/utils/color.ts` performs direct arithmetic (`-=`, `+=`) on these now-nullable properties.

## Fix

In `apps/web/src/utils/color.ts`, use nullish coalescing to default null coordinates to 0 before arithmetic:

```diff
- c.lch.l -= 5
+ c.lch.l = (c.lch.l ?? 0) - 5

- c.lch.l -= 20
- c.lch.c += 10
+ c.lch.l = (c.lch.l ?? 0) - 20
+ c.lch.c = (c.lch.c ?? 0) + 10
```

Defaulting to 0 is semantically correct — null lightness/chroma means achromatic/undefined, and 0 is the appropriate numeric baseline.

## Verification

- `pnpm --filter @gekichumai/dxrating-web build` passes after the fix
- Visually verify colored tag badges/overlays that use `deriveColor()`

--- a/apps/web/src/utils/color.ts
+++ b/apps/web/src/utils/color.ts
@@ -5,12 +5,12 @@ export const deriveColor = (color: string, type: 'border' | 'text' | 'overlay') 
   const c = new Color(color)
   return match(type)
     .with('border', () => {
-      c.lch.l -= 5
+      c.lch.l = (c.lch.l ?? 0) - 5
       return c.display()
     })
     .with('overlay', () => {
-      c.lch.l -= 20
-      c.lch.c += 10
+      c.lch.l = (c.lch.l ?? 0) - 20
+      c.lch.c = (c.lch.c ?? 0) + 10
       return c.display()
     })
     .with('text', () => {

